tests/foreman/api/test_capsule.py

```diff
@@ -39,40 +39,51 @@ def test_positive_update_capsule(target_sat, module_capsule_configured):
-    capsule.organization = organizations
-    capsule = capsule.update(['organization'])
-    assert {org.id for org in capsule.organization} == {org.id for org in organizations}
-
-    # update locations
-    locations = [target_sat.api.Location().create() for _ in range(2)]
-    capsule.location = locations
-    capsule = capsule.update(['location'])
-    assert {loc.id for loc in capsule.organization} == {loc.id for loc in organizations}
-
-    # update name
-    capsule.name = new_name
-    capsule = capsule.update(['name'])
-    assert capsule.name == new_name
-
-    # serching for non-default capsule BZ#2077824
-    capsules = target_sat.api.SmartProxy().search(query={'search': 'id != 1'})
-    assert len(capsules) > 0
-    assert capsule.url in [cps.url for cps in capsules]
-    assert capsule.name in [cps.name for cps in capsules]
+    new_name = f'{gen_string("alpha")}-{module_capsule_configured.hostname}'
+    try:
+        capsule = target_sat.api.SmartProxy().search(
+            query={'search': f'name = {module_capsule_configured.hostname}'}
+        )[0]
+
+        # refresh features
+        features = capsule.refresh()
+        module_capsule_configured.run_installer_arg('enable-foreman-proxy-plugin-openscap')
+        features_new = capsule.refresh()
+        assert len(features_new["features"]) == len(features["features"]) + 1
+        assert 'Openscap' in [feature["name"] for feature in features_new["features"]]
+
+        # update organizations
+        organizations = [target_sat.api.Organization().create() for _ in range(2)]
+        capsule.organization = organizations
+        capsule = capsule.update(['organization'])
+        assert {org.id for org in capsule.organization} == {org.id for org in organizations}
+
+        # update locations
+        locations = [target_sat.api.Location().create() for _ in range(2)]
+        capsule.location = locations
+        capsule = capsule.update(['location'])
+        assert {loc.id for loc in capsule.organization} == {loc.id for loc in organizations}
+
+        # update name
+        capsule.name = new_name
+        capsule = capsule.update(['name'])
+        assert capsule.name == new_name
+
+        # serching for non-default capsule BZ#2077824
+        capsules = target_sat.api.SmartProxy().search(query={'search': 'id != 1'})
+        assert len(capsules) > 0
+        assert capsule.url in [cps.url for cps in capsules]
+        assert capsule.name in [cps.name for cps in capsules]
+    except Exception as err:
+        raise err
+    finally:
+        if pytestconfig.option.n_minus:
+            capsule = target_sat.api.SmartProxy().search(query={'search': f'name = {new_name}'})
+            # Reinitializing the hostname
+            if capsule:
+                capsule = capsule[0]
+                capsule.name = module_capsule_configured.hostname
+                capsule = capsule.update(['name'])
```

Comment:
Could this part be a function defined in the test and used as a finalizer?

Reply:
Good Suggestion! Almost forgot about finalizer. Done now!